### ISSUE TITLE
1.14.7 release plan

### ISSUE THREAD

**Issue by patricklodder** [dogecoin/dogecoin#3407]
I learned today that there are multiple 3rd parties desperately waiting for a release that includes #3299. To facilitate, I'm proposing to start the release pressure cooker now. This means that we would **drastically reduce the scope** of what is still pending towards 1.14.7, and **do a release this month**.

Please read this proposal carefully.

## Proposed deadlines

- [x] Deadline for any pull requests that change UI strings: 2024-02-12 18:00 UTC
- [x] Deadline for any other pull requests and translations: 2024-02-25 18:00 UTC
- [x] Deadline for any tail-end maintainer work for release preparation: 2024-02-27 18:00 UTC

## Scope

For a [full list of in-scope PRs that are currently open](https://github.com/dogecoin/dogecoin/pulls?q=is%3Apr+milestone%3A1.14.7+is%3Aopen), you can filter the pull request view by `milestone:1.14.7`

### Must haves

- [x] #3389 - fee estimates are totally broken without
- important dogecoin-qt fixes
    - [x] #3412
    - [x] #3415
    - [x] #3420

### Low hanging fruit

Would be nice if this can be finished, but let’s not make any of it blocking:

- [x] #3388 - fixes #3352
- [x] #3414 - fixes #3072
- [x] #3297 - important functionality add, deeply tested, just needs a squash
- [x] #2676 - just needs a test
- [x] #3421 - removes warnings, minor
- [x] #3427 - bugfix that re-enables backward compatibility with qt4 
- [x] #3429 - small fix

### Documentation updates
- [x] macOS build: #2686
- [x] WSL build: #3411
- [x] List minimum versions in `INSTALL.md`: #3417
- [x] Fedora build: #3434
- [x] RPC maturity documentation: #3443

### Pushed forward, with deep regrets

These should get love first thing after release:

- #2459 - because we still don’t have a working compact block test, hard prerequisite.
- #3161 - no time to truly test this: time-intensive
- #3188 - no test results for some platforms yet, intensive
- enablers for #3149 - too much work for now
- #3337 - Would protect custom compiles w/ experimental, but there’s no code yet

**Push everything else, because they're not ready**

## String translation updates

- [x] `fr` & `fr_CA`: #3195
- [x] `zh_CN`: #3419
- [x] `it`: #3428
- [x] `ko_KR`: #3430
- [x] `pl`: #3431 
- [x] `nl`: #3435
- `ar`: #3439 - postponed due to missing language review.
- [x] `fr_CA`: #3441 

## Maintainer tasks for release

- [x] String freeze and translations: #3413
- [x] Update fixed seeds: #3422
- [x] Update checkpoints: #3416
- [x] Update manpages: #3423
- Update copyright notices
    - [x] copyright statement in `contrib/debian/copyright`: #3444 
    - [x] per-file copyright and `COPYING`: #3447
- Write release notes:
    - [x] initial draft: #3442 
- [x] Fixate v1.14.7 in source: #3448
- [x] Tag v1.14.7 - on 2024-02-27 latest
- [x] Gitian builds
- [x] Fast-fwd 1.14-maint
- [x] Fast-fwd master
- [x] Upload binaries and sha256sum files

### Post-release tasks

- [x] Create 1.15.0-dev branch
- [x] Open 1.15.0-dev - #3453 
- [x] Rebase open PRs
- [x] Delete 1.14.7-dev

**Comment by patricklodder** [comment]
Added #2676 to low hanging fruit - I'm honestly not sure why this is still open in the first place.

**Comment by slightlyskepticalpotat** [comment]
I'm able to help with translation strings for the various Chinese locales when time.

**Comment by georgeartem** [comment]
Following. Would be happy to draft up some release notes as needed. Going to dig in to each of these PRs next week🙏Sent from my iPhoneOn Feb 8, 2024, at 8:33 PM, Anthony Chen ***@***.***> wrote:﻿
I'm able to help with translation strings for the various Chinese locales when time.

**Comment by patricklodder** [comment]
Added #3411 - doc update for WSL

**Comment by patricklodder** [comment]
I've shifted the string freeze and final PR deadlines 1 day each because we're waiting on an important dogecoin-qt patch to be PR'd that changes strings. The shift does not change the planned release date: I'll personally simply work harder on the maintainer items.

**Comment by patricklodder** [comment]
> what # is the dogecoin-qt patch?

#3412 is the PR that was preventing string freeze.

**Comment by edtubbs** [comment]
Agreed on the release plan given the current situation. My runtime detection won't make it in this time, but no worries. Currently working on #3411, will build and test both 64 and 32-bit.

**Comment by georgeartem** [comment]
It appears github projects have been used for past releases. Would it be
helpful to wrap these deliverables into one? Some of these appear on the
"review-merge board" but not all.

**Comment by patricklodder** [comment]
> It appears github projects have been used for past releases.

Not really - we used board for milestones but they were kinda redundant. Do you think that doing so makes it clearer than the top post in this issue?

**Comment by georgeartem** [comment]
It’s kind of tomato potato… when trying to track down the status of the issues/prs in this thread getblockstats had already been merged and it took me some digging to find it.Sent from my iPhoneOn Feb 15, 2024, at 1:08 PM, Old Dip Tracker ***@***.***> wrote:﻿

It appears github projects have been used for past releases.

Not really - we used board for milestones but they were kinda redundant. Do you think that doing so makes it clearer than the top post in this issue?

**Comment by patricklodder** [comment]
> It’s kind of tomato potato… when trying to track down the status of the issues/prs in this thread getblockstats had already been merged and it took me some digging to find it

You can see it by the checkmark - I do my best to mark it as soon as I see something got merged - and the color of the PR icon (purple = merged, green is pending, grey = draft)

**Comment by georgeartem** [comment]
Gotcha, thanks for clarifying. Sent from my iPhoneOn Feb 15, 2024, at 1:54 PM, Old Dip Tracker ***@***.***> wrote:﻿

It’s kind of tomato potato… when trying to track down the status of the issues/prs in this thread getblockstats had already been merged and it took me some digging to find it

You can see it by the checkmark - I do my best to mark it as soon as I see something got merged - and the color of the PR icon (purple = merged, green is pending, grey = draft)

**Comment by georgeartem** [comment]
Thank you for doing that. Decentralized development practices are new to me. The type of thing I was suggesting is only really useful to help organize a full-time PM to assist with briefing “executive leadership” but since we don’t have any of that here it’s kind of up to the wisdom of the group. Appreciate you all 🙏Sent from my iPhoneOn Feb 16, 2024, at 5:09 AM, Old Dip Tracker ***@***.***> wrote:﻿
@georgeartem I cleaned up the 1.14.7 milestone, so that you can now also view the state of all PRs by querying for milestone:1.14.7

**Comment by patricklodder** [comment]
As the string fixation has not been merged yet (turns out buying only 1 day was overly optimistic), I'm buying translators a week more because we cannot possibly expect all translations to be done within 2 days. I'm assuming #3413 gets merged today (2/17), so I'm shifting:

translation deadline to 2/25
release date to 2/27 latest

Still this month, not as fast as I hoped we'd be.

**Comment by patricklodder** [comment]
Strings are frozen and we're ready for translators to do their magic!

### How to edit a translation using the Github editor (works best on desktop)

#### 1. Go to [the main page](https://github.com/dogecoin/dogecoin) of this repo
#### 2. Press the "fork" button

<img width="1250" alt="1 fork-button" src="https://github.com/dogecoin/dogecoin/assets/1410115/66cc80fc-b8dc-41de-ac22-2e851249ec45">

#### 3. UNCHECK the "copy master branch only" checkmark, and press "create fork"

<img width="733" alt="fork" src="https://github.com/dogecoin/dogecoin/assets/1410115/505554e2-a7b4-4be6-b0ef-8ab278603ee4">

#### 4. Wait a few seconds
#### 5. Select the `1.14.7-dev` branch

<img width="351" alt="select dev" src="https://github.com/dogecoin/dogecoin/assets/1410115/9d3b457e-4c8a-46f5-adc9-bb83f713e636">

#### 6. Click the branch selector again and create a working branch for yourself by typing the name you want, and clicking the "create branch" option

<img width="314" alt="create branch" src="https://github.com/dogecoin/dogecoin/assets/1410115/04c97f95-a014-4d06-8854-836ec7e3040d">


#### 7. Browse the files to `src/qt/locale` and select the translation for your language

<img width="359" alt="browse src" src="https://github.com/dogecoin/dogecoin/assets/1410115/e7db8387-3bba-4171-b728-5873e866b01a">
<img width="205" alt="browse qt" src="https://github.com/dogecoin/dogecoin/assets/1410115/d7ab0f39-3269-41a3-8dd5-12cc90fac922">
<img width="215" alt="browse locale" src="https://github.com/dogecoin/dogecoin/assets/1410115/972c1921-fa96-4cc3-b5ec-a148b36aba06">


#### 8. Click the edit button and make your edits

<img width="1416" alt="edit" src="https://github.com/dogecoin/dogecoin/assets/1410115/bd1d9fb4-e5af-435b-9743-b0d5019482e8">

#### 9. When you're done, press the "commit changes" button  

<img width="166" alt="commit-button" src="https://github.com/dogecoin/dogecoin/assets/1410115/565b504f-a577-4fd6-84c1-993356bdee19">

#### 10. Describe your changes and press the "commit changes" button on the form

<img width="474" alt="commit" src="https://github.com/dogecoin/dogecoin/assets/1410115/a81a017a-5707-4c3b-9f20-646d2146def2">

#### Rinse & repeat 8 to 10 until you think it's ready.

----

### How to SUBMIT a translation when you're done

#### 11. Go to the root of your repo (`https://github.com/<yourusername>/dogecoin`)
#### 12. Make sure you are on the branch you created at step 6
#### 13. Click "contribute" and then "open pull request"

<img width="282" alt="Open pull" src="https://github.com/dogecoin/dogecoin/assets/1410115/023ddb3f-8b13-4a2b-9bb0-3bae92b380c5">

#### 14. IMPORTANT: Change the base branch to 1.14.7-dev
#### 15. Describe your pull request, and press the "open pull request" button.

<img width="923" alt="edit PR" src="https://github.com/dogecoin/dogecoin/assets/1410115/d78de7dd-0964-483a-a2f6-2e96f547f1e9">

-----

Several automated checks will run on your pull request, including a check if your translation file is syntactically good. Other people will help you get your translation in shape and maintainers can help you make a clean contribution where needed.

GOOD LUCK!

**Comment by patricklodder** [comment]
Per discussion in #3418, I have changed the next branch to be 1.15.0-dev. This will allow us to target more modern platforms and repair issues like #3356, while stopping to do crazy things like 99cbe512 and #3415.

**Comment by patricklodder** [comment]
1.14.7 is now frozen for all pull requests - except tail-end maintenance work (which is only the copyright update, any updates to release notes and version fixation). 

Open pull requests with a 1.14.7 milestone and mentioned above can of course still be reviewed and merged, as long as we see ACKs over the next 48h - maintainers can speed-merge anything that is non-controversial, if needed.

**Comment by patricklodder** [comment]
I've removed the Arabic translaslation from this release target because there was no language review in time - this can be picked up on a more relaxed timeframe for the next release. 

Have also marked #3442 and #3448 as urgent, as they've been open for a longer time now, to not hold up the release.

Thanks everyone for great work! It's been an amazing pressure cooker ❤️

**Comment by patricklodder** [comment]
Tag done: https://github.com/dogecoin/dogecoin/releases/tag/v1.14.7

I'm starting my gitian build and will report shasums here.

**Comment by chromatic** [comment]
My Gitian checksums, from `buildgitian https://github.com/dogecoin/dogecoin/ v1.14.7`:

```
cdfd75a7757493e386265a0c65614b2c6062d1f5f7177bddcc2eec00bea9480b  dogecoin-1.14.7-osx-unsigned.dmg
20439109506a9de9e8208b238c92c38c5f6554378f26a7660313b00d3e77f371  dogecoin-1.14.7-osx-unsigned.tar.gz
815884d43b8d386a70748626471935e83b3d794bfeea74318cdac4cbec2bae0a  dogecoin-1.14.7-osx64.tar.gz
8757ad2922a97e106a98835ce4fe0b7a43c7de5d5d1373f607cb89db3878dc35  src/dogecoin-1.14.7.tar.gz
2f9a58d8583af9c5623d47255d32df3bb694cf938a6fe03166bb56c73f162018  dogecoin-osx-1.14-res.yml
```

**Comment by chromatic** [comment]
That's two matching sets. Can we get a third?

**Comment by patricklodder** [comment]
> Can we get a third?

I'm starting to prep the notarization of the macOs dmg now, because that's still a manual process under my Apple developer key, like last time (hopefully). A third match would make these binaries shippable.

**Comment by alamshafil** [comment]
I can do the third build. I'll update once I get it working.

**Comment by slightlyskepticalpotat** [comment]
I'm also trying to create the third build, but I've never done this before and I'm not sure if it'll work.

**Comment by slightlyskepticalpotat** [comment]
I repeatedly error out here, could it be that my configuration of LXC is incorrect?
```
Compiling 1.14.7 osx
Initialized empty Git repository in /home/anthony/Downloads/dogecoin/contrib/gitian-builder/inputs/dogecoin/.git/
remote: Enumerating objects: 84628, done.
remote: Counting objects: 100% (408/408), done.
remote: Compressing objects: 100% (326/326), done.
remote: Total 84628 (delta 146), reused 327 (delta 82), pack-reused 84220
Receiving objects: 100% (84628/84628), 61.50 MiB | 10.76 MiB/s, done.
Resolving deltas: 100% (63688/63688), done.
From https://github.com/dogecoin/dogecoin
 * tag                   v1.14.7    -> FETCH_HEAD
--- Building for focal amd64 ---
Stopping target if it is up
Making a new image copy
sudo: unable to resolve host gitian: Temporary failure in name resolution
Starting target
Checking if target is up..............................
sudo: unknown user: ubuntu
sudo: unable to initialize policy plugin
```

**Comment by patricklodder** [comment]
> could it be that my configuration of LXC is incorrect?

No, it's the version. Gitian is EOL. LXC gitian only works on a bionic host due to breaking changes to debootstrap and you have to pre-download binaries too. I only run it on there because it's the best way to detect any prefix-jailbreak (like we found with xkb) because the image is created differently than docker.

I'd recommend running a docker build for now unless you have a bionic host ready to use.

**Comment by slightlyskepticalpotat** [comment]
Ah, now I feel stupid—I even remember reading https://github.com/dogecoin/dogecoin/issues/3095 around the last release.

**Comment by patricklodder** [comment]
No need to feel that way! What used to be the simplest method is now an extremely complex monstrosity.

Luckily we will soon start a new major and I'll take it upon myself to work on guix. Hopefully we'll be able to only need gitian 3 more times at most.

**Comment by slightlyskepticalpotat** [comment]
I'm having other issues with docker now, from searching online it seems to be a problem with apt-cacher configuration
```
./bin/gbuild:23:in `system!': failed to run on-target setarch x86_64 bash -x < var/build-script > var/build.log 2>&1 (RuntimeError)
```
Should my `MIRROR_HOST` be set to anything special? Right now it's just `127.0.0.1`.

**Comment by chromatic** [comment]
I think that's what I set `MIRROR_HOST` to. I had to enable the `apt-cacher-ng` service on the host as well.

**Comment by slightlyskepticalpotat** [comment]
The service is definitely running, and I even restarted and tried again to make sure, hmm.

**Comment by slightlyskepticalpotat** [comment]
The specific error in the `build.log` is 
```
+ python3 -m pip install /home/ubuntu/build/lief-0.12.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
WARNING: Requirement '/home/ubuntu/build/lief-0.12.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl' looks like a filename, but the file does not exist
Processing /home/ubuntu/build/lief-0.12.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
ERROR: Could not install packages due to an EnvironmentError: [Errno 2] No such file or directory: '/home/ubuntu/build/lief-0.12.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl'
```

**Comment by patricklodder** [comment]
Re: lief.

You need to get `https://depends.dogecoincore.org/lief-0.12.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl` and put it in `gitian-builder/inputs` together with the SDK and osslsigner files.

**Comment by slightlyskepticalpotat** [comment]
Thanks! I grabbed the file from another source in the meantime but the checksums line up.

The build process is humming along now and I'll edit this message with my checksums,
```
cdfd75a7757493e386265a0c65614b2c6062d1f5f7177bddcc2eec00bea9480b  dogecoin-1.14.7-osx-unsigned.dmg
20439109506a9de9e8208b238c92c38c5f6554378f26a7660313b00d3e77f371  dogecoin-1.14.7-osx-unsigned.tar.gz
815884d43b8d386a70748626471935e83b3d794bfeea74318cdac4cbec2bae0a  dogecoin-1.14.7-osx64.tar.gz
8757ad2922a97e106a98835ce4fe0b7a43c7de5d5d1373f607cb89db3878dc35  src/dogecoin-1.14.7.tar.gz
dd7a88445223e20d3e6ed456730a99c73d9870c2d52138944396a2204b190561  dogecoin-osx-1.14-res.yml
```
After everything completes, should I make a PR to https://github.com/dogecoin/gitian.sigs?

**Comment by alamshafil** [comment]
Here are my Gitian checksums using Docker.  Will update as it builds.

```
cdfd75a7757493e386265a0c65614b2c6062d1f5f7177bddcc2eec00bea9480b  dogecoin-1.14.7-osx-unsigned.dmg
20439109506a9de9e8208b238c92c38c5f6554378f26a7660313b00d3e77f371  dogecoin-1.14.7-osx-unsigned.tar.gz
815884d43b8d386a70748626471935e83b3d794bfeea74318cdac4cbec2bae0a  dogecoin-1.14.7-osx64.tar.gz
8757ad2922a97e106a98835ce4fe0b7a43c7de5d5d1373f607cb89db3878dc35  src/dogecoin-1.14.7.tar.gz
```

**Comment by slightlyskepticalpotat** [comment]
All checksums match, PR created, thanks @patricklodder and @chromatic for the support!

**Comment by patricklodder** [comment]
I have merged @slightlyskepticalpotat's gitian sigs and finalized the release as draft, including notarized macOs binaries.

Here's the SHA256SUMS.asc:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

b8fb8050b19283d1ab3c261aaca96d84f2a17f93b52fcff9e252f390b0564f31  dogecoin-1.14.7-aarch64-linux-gnu.tar.gz
5606ccf343a403797518072e4af4c693f1b1f826e4fb58ea1b9f3335828c83a6  dogecoin-1.14.7-arm-linux-gnueabihf.tar.gz
4c5c19fc7c561699947a9fd0bce0e037e6da767f6275c6db9eb6e36c2e395a79  dogecoin-1.14.7-i686-pc-linux-gnu.tar.gz
cbdd14693e6e4e81d53c034ef05d806495f00d7946c3c1463a5e1550629cf635  dogecoin-1.14.7-osx-signed.dmg
cdfd75a7757493e386265a0c65614b2c6062d1f5f7177bddcc2eec00bea9480b  dogecoin-1.14.7-osx-unsigned.dmg
20439109506a9de9e8208b238c92c38c5f6554378f26a7660313b00d3e77f371  dogecoin-1.14.7-osx-unsigned.tar.gz
815884d43b8d386a70748626471935e83b3d794bfeea74318cdac4cbec2bae0a  dogecoin-1.14.7-osx64.tar.gz
1679563488b7145b027a6801b0d3e458a012826b1447295119f94d7c615b5e5c  dogecoin-1.14.7-win-unsigned.tar.gz
738e70d06263e20f06304fbbd1b0e1481a7d0da3434b675e4b1e680644c09b8a  dogecoin-1.14.7-win32-setup-unsigned.exe
d870d08ea59a6f40b3f2a03048bcdcfc4150896430ffade1bc466925dda93737  dogecoin-1.14.7-win32.zip
e1bc8be88d22af22bb26c2d87cfe497a808fc54817921ced416392a1ed78c0a2  dogecoin-1.14.7-win64-setup-unsigned.exe
8475ba266761e830f2a270c542b5513a3e33892951e8d31318ca3b48cf5465f1  dogecoin-1.14.7-win64.zip
9cd22fb3ebba4d407c2947f4241b9e78c759f29cdf32de8863aea6aeed21cf8b  dogecoin-1.14.7-x86_64-linux-gnu.tar.gz
-----BEGIN PGP SIGNATURE-----

iQEzBAEBCAAdFiEE3G70qL+fGx5N4e5SLTo0W5jQ3B8FAmXfVsAACgkQLTo0W5jQ
3B9kLQf/Wlaym2nnmynUqGf9IbdNLK5K0Uj6Ktl0XKDp5rISnM6lSdrjRxyNTYWY
nB5fCXx31x7Sc/60jryUW4ysbxioGlfGmsqiVNs1GGAqBVRZyc0hgKtGMG7MlHf8
omWn7pQcZN5wY5eso86v2IC0n/3OhzRA2LxVr76dUbs4WuNaodxjcQGtJ4XMaLnr
dW9KQxwFKg+taOPnufY+isy9SqGGJWqdQ5R4vX4gKbyFPCghhhPiI0VNBPpQ7HLu
iS+6xPI/8KfjzzbRH9pN8OWFp/1fCylOYkb7sadjZwZPz3pL9PZglNQluzk2KFZG
WDoRSJdEhodqIgnOPqQvhWbo0T/p9Q==
=vEm2
-----END PGP SIGNATURE-----
```

If no objections, I'll push branches and make the binaries public in 2h from now.

**Comment by slightlyskepticalpotat** [comment]
No objections here, looks the same as mine.

**Comment by patricklodder** [comment]
DONE! 🥳

**Comment by FierceSkit** [comment]
> DONE! 🥳

Congrats!
Also to @chromatic and all the others!
